code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -990,6 +990,8 @@
 	// Remove upgrades.
 	for(var/obj/item/I in upgrades)
 		I.forceMove(get_turf(src))
+		if (hascall(I,"dropped"))
```

Comment:
Instead of this hack, just call `deactivate()` on the upgrades

Reply:
Thing is, the selfrepair/dropped() proc (in robot_upgrades.dm) has several other things it does first before calling deactivate_sr() (dropped leads to check_dropped, which eventually leads to deactivate_sr). Should I just move all that over to deactivate_sr()? 

Additionally (and I don't mean this as an attack or insult) how is calling deactivate_sr() less hacky than calling dropped()? Either way, it's an extra call that currently only fixes a single edge case, right?

Reply:
These are just wrappers, there's no change needed in the repair itself.

Reply:
I still don't think I'm super clear on what you were suggesting. Neither `deactivate()` nor `deactivate_sr()` compile when swapped for `dropped()`, so I might be misunderstanding.

Removed the unneeded check, though

Reply:
You need change the iterated path to `/obj/item/borg/upgrade`, and switch the call to `deactivate(src)`, sorry for late response.